client/me/application-passwords/index.jsx

```diff
@@ -3,106 +3,85 @@
 /**
  * External dependencies
  */
-
-import React from 'react';
-import createReactClass from 'create-react-class';
-import { localize } from 'i18n-calypso';
-import debugFactory from 'debug';
-const debug = debugFactory( 'calypso:application-passwords' );
+import React, { Component } from 'react';
+import classNames from 'classnames';
+import Gridicon from 'gridicons';
 import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
-/* eslint-disable no-restricted-imports */
-import observe from 'lib/mixins/data-observe';
-/* eslint-enable no-restricted-imports */
 import AppPasswordItem from 'me/application-password-item';
-import notices from 'notices';
-import SectionHeader from 'components/section-header';
 import Button from 'components/button';
-import Gridicon from 'gridicons';
-import FormFieldset from 'components/forms/form-fieldset';
-import FormTextInput from 'components/forms/form-text-input';
-import FormLabel from 'components/forms/form-label';
+import Card from 'components/card';
 import FormButton from 'components/forms/form-button';
 import FormButtonsBar from 'components/forms/form-buttons-bar';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
 import FormSectionHeading from 'components/forms/form-section-heading';
-import Card from 'components/card';
-import classNames from 'classnames';
-import { errorNotice } from 'state/notices/actions';
+import FormTextInput from 'components/forms/form-text-input';
+import QueryApplicationPasswords from 'components/data/query-application-passwords';
+import SectionHeader from 'components/section-header';
+import {
+	clearNewApplicationPassword,
+	createApplicationPassword,
+} from 'state/application-passwords/actions';
+import { getApplicationPasswords, getNewApplicationPassword } from 'state/selectors';
 import { recordGoogleEvent } from 'state/analytics/actions';
 
-const ApplicationPasswords = createReactClass( {
-	displayName: 'ApplicationPasswords',
-	mixins: [ observe( 'appPasswordsData' ) ],
-
-	componentDidMount: function() {
-		debug( this.displayName + ' React component is mounted.' );
-	},
+class ApplicationPasswords extends Component {
+	initialState = {
+		applicationName: '',
+		addingPassword: false,
+		submittingForm: false,
+	};
 
-	componentWillUnmount: function() {
-		debug( this.displayName + ' React component is unmounting.' );
-	},
+	state = this.initialState;
 
-	getInitialState: function() {
-		return {
-			applicationName: '',
-			addingPassword: false,
-			submittingForm: false,
-		};
-	},
+	componentWillReceiveProps( nextProps ) {
+		if ( this.state.submittingForm && ! this.props.newAppPassword && !! nextProps.newAppPassword ) {
+			this.setState( { submittingForm: false } );
```

Comment:
I'll try to read this condition:

1. We are submitting
1. We don't have a new app password (that would hide the form)
1. We aren't going to receive a new app password

Should we `setState( this.initialState )` like below? Perhaps dedicated `clearState` method?

Reply:
> 1. We are submitting

✅ 

> 2. We don't have a new app password (that would hide the form)

✅ 

> 3. We aren't going to receive a new app password

❌ We are going to receive a new app password

And regarding your suggestion - I'll leave it as-is, this PR is already introducing a bunch of changes, while trying to keep the behavior as consistent as possible to what we had before.